src/core/server/server.api.md

```diff
@@ -248,6 +244,19 @@ export interface HttpServiceStart {
     isListening: (port: number) => boolean;
 }
 
+// @public
+export interface IContextContainer<TContext extends {}, THandlerReturn, THandlerParameters extends any[] = []> {
+    // Warning: (ae-forgotten-export) The symbol "Promisify" needs to be exported by the entry point index.d.ts

```

Comment:
`Promisify` helps avoid nested Promise types (which I was surprised are even representable in TypeScript, seems like a bug to me).

If I just used `Promise<THandlerReturn>` the return type would resolve to `Promise<Promise<THandlerReturn>>` if `THandlerReturn` is a Promise. This would still work with `await` and what not, but results in unexpected types in some cases. For instance, `UnwrapPromise<Promise<Promise<T>>` resolves to `Promise<T>` rather than just `T`.

I can add it as an export with documentation explaining what it does, alternatively I could inline the conditional type as part of this interface directly.

Reply:
This type can also be simplified to `T extends Promise<any> ? T : Promise<T>`